src/helpers/spaces.ts

```diff
@@ -219,6 +219,53 @@ async function getVotes(): Promise<Record<string, { votesCount7d: number }>> {
   );
 }
 
+export async function getCombinedMembersAndVoters(
+  spaceId: string,
+  cursor: string | null,
+  pageSize: number,
+  knownAdmins: string[] = [],
+  knownModerators: string[] = [],
+  knownMembers: string[] = []
+) {
+  const params: (string | number)[] = [spaceId];
+  const exclusionList = [...knownAdmins, ...knownModerators, ...knownMembers];
+
+  // Other roles are already known and fetched at the app level while Space Verification
+  // Building the exclusion clause only if the exclusion list is not empty
+  let exclusionClause = '';
+  if (exclusionList.length > 0) {
+    const placeholders = exclusionList.map(() => '?').join(', ');
+    exclusionClause = `AND voter NOT IN (${placeholders})`;
```

Comment:
Can't this be used to inject arbitrary SQL? I think `placeholders` should be added as params so it get parsed.

Reply:
I don't think that would be of concern here because 
1. Placeholders values are not user input values
2. Placeholder values are either empty lists or space.admins, space.moderators, space.members
3. `params.push(...exclusionList)`  This pushes the actual exclusion list values (the voters you want to exclude) into the params array, which is passed to the query execution.
4. ` const results =await db.queryAsync(query, params);`  The database library takes care of escaping the values in params, preventing SQL injection.